dart/constraint/ConstrainedGroup.h

```diff
@@ -107,7 +108,7 @@ class ConstrainedGroup
   std::vector<ConstraintBase*> mConstraints;
 
   ///
-  dynamics::Skeleton* mRootSkeleton;
+  std::shared_ptr<dynamics::Skeleton> mRootSkeleton;
```

Comment:
That sounds good to me, although I'm pretty sure SkeletonPtr and ShapePtr are the only types that would be able to go in there, because BodyNodePtr operates in such a way that it needs to have the BodyNode class fully defined before it can be instantiated, and JointPtr depends directly on BodyNodePtr. So I'd be a little concerned that an end user would see "DynamicsTypes.h" and assume that they can get BodyNodePtr and JointPtr from it as well, when they actually can't.